components/api/src/main/java/com/hotels/styx/api/messages/StreamingHttpResponse.java

```diff
@@ -0,0 +1,448 @@
+import static com.hotels.styx.api.HttpHeaderValues.CHUNKED;
+import static io.netty.buffer.Unpooled.compositeBuffer;
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
+import static io.netty.util.ReferenceCountUtil.release;
+import static java.lang.Integer.parseInt;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * HTTP response with a fully aggregated/decoded body.
+ */
+public class StreamingHttpResponse implements StreamingHttpMessage {
+    private final HttpVersion version;
+    private final HttpResponseStatus status;
+    private final HttpHeaders headers;
+    private final Observable<ByteBuf> body;
+    private final List<HttpCookie> cookies;
+
+    StreamingHttpResponse(Builder builder) {
+        this.version = builder.version;
+        this.status = builder.status;
+        this.headers = builder.headers.build();
+        this.body = builder.body;
+        this.cookies = ImmutableList.copyOf(builder.cookies);
+    }
+
+    /**
+     * Creates an HTTP response builder with a status of 200 OK and empty body.
+     *
+     * @return a new builder
+     */
+    public static Builder response() {
+        return new Builder();
+    }
+
+    /**
+     * Creates an HTTP response builder with a given status and empty body.
+     *
+     * @param status response status
+     * @return a new builder
+     */
+    public static Builder response(HttpResponseStatus status) {
+        return new Builder(status);
+    }
+
+    /**
+     * Creates an HTTP response builder with a given status and body.
+     *
+     * @param status response status
+     * @param body response body
+     * @return a new builder
+     */
+    public static Builder response(HttpResponseStatus status, Observable<ByteBuf> body) {
+        return new Builder(status).body(body);
+    }
+
+    @Override
+    public Optional<String> header(CharSequence name) {
+        return headers.get(name);
+    }
+
+    @Override
+    public List<String> headers(CharSequence name) {
+        return headers.getAll(name);
+    }
+
+    @Override
+    public HttpHeaders headers() {
+        return headers;
+    }
+
+    @Override
+    public List<HttpCookie> cookies() {
+        return cookies;
+    }
+
+    @Override
+    public Observable<ByteBuf> body() {
+        return body;
+    }
+
+    @Override
+    public HttpVersion version() {
+        return version;
+    }
+
+    public Builder newBuilder() {
+        return new Builder(this);
+    }
+
+    public HttpResponseStatus status() {
+        return status;
+    }
+
+    public boolean isRedirect() {
+        return status().code() >= 300 && status().code() < 400;
+    }
+
+    public <T> Observable<FullHttpResponse<T>> toFullHttpResponse(Function<ByteBuf, T> decoder, int maxContentBytes) {
```

Comment:
Update as per latest `FullHttpResponse` without `T` type parameter. Essentially, the `decoder` parameter ought to be removed as unnecessary.